spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/ParquetReaderType.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark;
+
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/** Enumerates the types of Parquet readers. */
+public enum ParquetReaderType {
+  /** ICEBERG type utilizes the built-in Parquet reader. */
+  ICEBERG("iceberg"),
+
+  /**
+   * COMET type changes the Parquet reader to the Apache DataFusion Comet Parquet reader. Comet
+   * Parquet reader performs I/O and decompression in the JVM but decodes in native to improve
+   * performance. Additionally, Comet will convert Spark's physical plan into a native physical plan
+   * and execute this plan natively.
+   *
+   * <p>TODO: Implement {@link org.apache.comet.parquet.SupportsComet} in SparkScan to convert Spark
+   * physical plan to native physical plan for native execution.
+   */
+  COMET("comet");
+
+  private final String parquetReaderType;
+
+  ParquetReaderType(String readerType) {
+    this.parquetReaderType = readerType;
+  }
+
+  public static ParquetReaderType fromName(String parquetReaderType) {
+    Preconditions.checkArgument(parquetReaderType != null, "Parquet reader type is null");
+
+    if (ICEBERG.parquetReaderType().equalsIgnoreCase(parquetReaderType)) {
+      return ICEBERG;
+
+    } else if (COMET.parquetReaderType().equalsIgnoreCase(parquetReaderType)) {
+      return COMET;
+
+    } else {
+      throw new IllegalArgumentException("Unknown parquet reader type: " + parquetReaderType);
+    }
+  }
+
+  public String parquetReaderType() {
```

Comment:
I am not sure calling the field and method as `parquetReaderType` is a good idea. I'd either consider something shorter like `typeName`/`identifier`/`key` or even something like below:

```
public static ParquetReaderType fromString(String typeAsString) {
  Preconditions.checkArgument(typeAsString != null, "Parquet reader type is null");
  try {
    return ParquetReaderType.valueOf(typeAsString.toUpperCase());
  } catch (IllegalArgumentException e) {
    throw new IllegalArgumentException("Unknown parquet reader type: " + typeAsString);
  }
}
```

Reply:
I do like the snippet a bit more, it is shorter. What do you think, @huaxingao?